src/Polly.Contrib.DuplicateRequestCollapser.RedisDistributedLock/StackExchangeRedisDistributedLockProvider.cs

```diff
@@ -0,0 +1,170 @@
+﻿#nullable enable
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+using StackExchange.Redis;
+
+namespace Polly.Contrib.DuplicateRequestCollapser.RedisDistributedLock
+{
+    /// <summary>
+    /// Provides a lock scoped to a distributed key. Unlike other local locks, this lock provider
+    /// can throw an <see cref="OperationCanceledException"/> if the lock attempt times out.
+    /// </summary>
+    public class StackexchangeRedisDistributedLockProvider : ISyncLockProvider
+    {
+        private readonly StackexchangeRedisDistributedLockProviderOptions options;
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="options">Options</param>
```

Comment:
fuller intellisense, for a publicly-released OSS project?

Reply:
Added more comments :)